ui/src/app/core/auth/error.inerceptor.ts

```diff
@@ -19,23 +19,31 @@ import { Observable, throwError } from 'rxjs';
 import { catchError } from 'rxjs/operators';
 
 import { AuthService } from './auth.service';
+import { Routes } from '../../data/enums/routers-url.enum';
+import { Router } from '@angular/router';
 
 @Injectable()
 export class ErrorInterceptor implements HttpInterceptor {
   private authService: AuthService;
 
-  constructor(private injector: Injector) {
+  constructor(private injector: Injector, private router: Router) {
     this.authService = this.injector.get(AuthService);
   }
 
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     return next.handle(request).pipe(
       catchError(
         (response: any): Observable<HttpEvent<any>> => {
-          if (response instanceof HttpErrorResponse && response.status === 401) {
-            this.authService.logOut();
+          if (response instanceof HttpErrorResponse) {
+            switch (response.status) {
+              case 401:
+                this.authService.logOut();
+                break;
+              case 400:
```

Comment:
is this a requirement to navigate to home on every 400 request? Maybe you wanted to put 404 here?

Reply:
For example. I have a link where I try to log in to my account. I will be redirected to the login page, after successfully logging in, I will return to the link, but if I log in to another account and am redirected to the link, I will get the error "400".